rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,431 @@
+        if credentials.api_key is not None:
+            self.session.headers.update({'x-api-key': credentials.api_key})
+        return changed
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> ExchangeQueryBalances:
+        """Return the account balances on Woo """
+        response = self._api_query('v3/balances')
+
+        if response.status_code != HTTPStatus.OK:
+            result, msg = self._process_unsuccessful_response(
+                response=response,
+                case='balances',
+            )
+            return result, msg
+        try:
+            response_dict = jsonloads_dict(response.text)
+        except JSONDecodeError as e:
+            msg = f'Woo returned invalid JSON response: {response.text}.'
+            log.error(msg)
+            raise RemoteError(msg) from e
+        account_balances = self._deserialize_accounts_balances(response_dict=response_dict)
+        return account_balances, ''
+
+    def _deserialize_accounts_balances(
+            self,
+            response_dict: dict[str, Any],
+    ) -> dict[AssetWithOracles, Balance]:
+        try:
+            balances = response_dict['data']['holding']
+        except KeyError as e:
+            msg = 'Woo balances JSON response is missing data key'
+            log.error(msg, response_dict)
+            raise RemoteError(msg) from e
+
+        assets_balance: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        for entry in balances:
+            symbol = entry.get('token')
+            try:
+                amount = deserialize_asset_amount(entry.get('holding') + entry.get('staked'))
+                if amount == ZERO:
+                    continue
+                asset = asset_from_woo(symbol)
+            except DeserializationError as e:
+                log.error(
+                    'Error processing a Woo balance.',
+                    entry=entry,
+                    error=str(e),
+                )
+                self.msg_aggregator.add_error(
+                    'Failed to deserialize a Woo balance. '
+                    'Check logs for details. Ignoring it.',
+                )
+                continue
+            except (UnknownAsset, UnsupportedAsset) as e:
+                log.error(str(e))
+                asset_tag = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                self.msg_aggregator.add_warning(
+                    f'Found {asset_tag} Woo asset {e.identifier}. Ignoring its balance query.',
+                )
+                continue
+            try:
+                usd_price = Inquirer().find_usd_price(asset=asset)
+            except RemoteError as e:
+                log.error(str(e))
+                self.msg_aggregator.add_error(
+                    f'Error processing Woo balance result due to inability to '
+                    f'query USD price: {e!s}. Skipping balance entry.',
```

Comment:
```suggestion
                    f'query USD price: {e}. Skipping balance entry.',
```

I believe that the !s is not need. Can you check?